src/carts/src/main/java/mushop/carts/repositories/CartRepositoryDatabaseImpl.java

```diff
@@ -1,66 +1,86 @@
-import java.util.Properties;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-import java.io.InputStream;
-
-import javax.json.Json;
-import javax.json.bind.Jsonb;
-import javax.json.bind.JsonbBuilder;
-
-import io.helidon.config.Config;
+package mushop.carts.repositories;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.event.StartupEvent;
+import io.micronaut.runtime.event.annotation.EventListener;
+import mushop.carts.entitites.Cart;
 import oracle.jdbc.OracleConnection;
 import oracle.soda.OracleCollection;
 import oracle.soda.OracleCursor;
 import oracle.soda.OracleDatabase;
 import oracle.soda.OracleDocument;
 import oracle.soda.rdbms.OracleRDBMSClient;
 import oracle.ucp.jdbc.PoolDataSource;
-import oracle.ucp.jdbc.PoolDataSourceFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.json.Json;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
 
 /**
  * Implements CartRepository using Oracle Database JSON collections. SODA is the
  * simple CRUD-based API that allows the application to interact with document
  * collections in the autonomous database.
  */
+@Singleton
+@Primary
+@Requires(env = Environment.ORACLE_CLOUD)
 public class CartRepositoryDatabaseImpl implements CartRepository {
 
-    /** Factory for SODA (simple oracle document access) api */
+    public static final Logger LOG = LoggerFactory.getLogger(CartRepositoryDatabaseImpl.class);
+
+    /**
+     * Factory for SODA (simple oracle document access) api
+     */
     private static final OracleRDBMSClient SODA;
 
-    /** The name of the backing collection */
-    private final String collectionName;
+    static {
+        // Optimization: cache collection metadata to avoid extra roundtrips
+        // to the database when opening a collection
+        Properties props = new Properties();
+        props.put("oracle.soda.sharedMetadataCache", "true");
+        SODA = new OracleRDBMSClient(props);
+    }
 
-    /** Pool of reusable database connections */
-    protected PoolDataSource pool;
+    /**
+     * The name of the backing collection
+     */
+    @Value("${carts.collection}")
+    private String collectionName;
 
-    /** Used to automatically convert a Cart object to and from JSON */
-    private Jsonb jsonb;
+    /**
+     * Pool of reusable database connections
+     */
+    @Inject
+    protected PoolDataSource pool;
 
-    private final static Logger log = Logger.getLogger(CartService.class.getName());
+    /**
+     * Used to automatically convert a Cart object to and from JSON
+     */
+    @Inject
+    private ObjectMapper objectMapper;
```

Comment:
Switch this to private final fields and use constructor injection instead of `@Inject`